cartography/intel/gcp/compute.py

```diff
@@ -147,6 +147,147 @@ def sync_compute_disks(
+    try:
+        req = compute.targetSslProxies().list(project=project_id)
+        while req is not None:
+            res = req.execute()
+            if 'items' in res:
+                ssl_proxies.extend(res['items'])
+            req = compute.targetSslProxies().list_next(previous_request=req, previous_response=res)
+
+        if common_job_parameters.get('pagination', {}).get('compute', None):
+            pageNo = common_job_parameters.get("pagination", {}).get("compute", None)["pageNo"]
+            pageSize = common_job_parameters.get("pagination", {}).get("compute", None)["pageSize"]
+            totalPages = len(ssl_proxies) / pageSize
+            if int(totalPages) != totalPages:
+                totalPages = totalPages + 1
+            totalPages = int(totalPages)
+            if pageNo < totalPages or pageNo == totalPages:
+                logger.info(f'pages process for compute ssl proxies {pageNo}/{totalPages} pageSize is {pageSize}')
+            page_start = (common_job_parameters.get('pagination', {}).get('compute', None)[
+                          'pageNo'] - 1) * common_job_parameters.get('pagination', {}).get('compute', None)['pageSize']
+            page_end = page_start + common_job_parameters.get('pagination', {}).get('compute', None)['pageSize']
+            if page_end > len(ssl_proxies) or page_end == len(ssl_proxies):
+                ssl_proxies = ssl_proxies[page_start:]
+            else:
+                has_next_page = True
+                ssl_proxies = ssl_proxies[page_start:page_end]
+                common_job_parameters['pagination']['compute']['hasNextPage'] = has_next_page
+
+        return ssl_proxies
+
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+            logger.warning(
+                (
+                    "Could not retrieve ssl proxies on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            raise
+
+@timeit
+def transform_ssl_proxies(proxies: List, project_id: str) -> List[Resource]:
+    list_proxies = []
+    for proxy in proxies:
+        proxy['id'] = f"projects/{project_id}/global/targetSslProxies/{proxy['name']}"
+        list_proxies.append(proxy)
+
+    return list_proxies
+
+@timeit
+def load_proxies(session: neo4j.Session, proxies: List[Dict], project_id: str, update_tag: int) -> None:
+    session.write_transaction(load_proxies_tx, proxies, project_id, update_tag)
+
+@timeit
+def load_proxies_tx(
+    tx: neo4j.Transaction, proxies: List[Dict],
+    project_id: str, gcp_update_tag: int,
+) -> None:
+
+    query = """
+    UNWIND {Proxies} as p
+    MERGE (proxy:GCPProxy{id:p.id})
+    ON CREATE SET
+        proxy.firstseen = timestamp()
+    SET
+        proxy.lastupdated = {gcp_update_tag},
```

Comment:
We should also capture a type field to store whether it's a https proxy or ssl proxy. values can be `https` or `ssl`